presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -464,436 +477,595 @@ public void testWithJoinTables()
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
+    }
+
+    @Test
+    public void singleSubquerySingleCompatibleView()
+    {
+        String subquery = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String originalViewSql = subquery;
+
+        String baseQuerySql = format("SELECT a, b FROM (%s)", subquery);
+        String expectedRewrittenSql = format("SELECT a, b FROM (SELECT a, b FROM (%s))", VIEW_1);
+
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, originalViewSql);
+    }
+
+    @Test
+    public void singleSubquerySingleIncompatibleView()
+    {
+        String subquery = format("SELECT a, b, c FROM %s", BASE_TABLE_1);
+        String originalViewSql = format("SELECT a, b FROM %s", BASE_TABLE_1);
+
+        String baseQuerySql = format("SELECT a, b FROM (%s)", subquery);
+
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, originalViewSql);
+    }
+
+    @Test
+    public void multipleViewsSameBaseTableAllCompatible()
+    {
+        String subquery1 = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String subquery2 = format("SELECT c FROM %s", BASE_TABLE_1);
+        String viewSql1 = subquery1;
+        String viewSql2 = subquery2;
+
+        String baseQuerySql = format("SELECT a, b, c FROM (%s) UNION ALL (%s)", subquery1, subquery2);
+        String expectedRewrittenSql = format("SELECT a, b, c FROM " +
+                "(SELECT a, b FROM (%s)) " +
+                "UNION ALL " +
+                "(SELECT c FROM (%s))",
+                VIEW_1, VIEW_2);
+
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, viewSql1, viewSql2);
+    }
+
+    @Test
+    public void multipleViewsSameBaseTableNoneCompatible()
+    {
+        String subquery1 = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String subquery2 = format("SELECT b, c FROM %s", BASE_TABLE_1);
+        String viewSql1 = format("SELECT a FROM %s", BASE_TABLE_1);
+        String viewSql2 = format("SELECT b FROM %s", BASE_TABLE_1);
+
+        String baseQuerySql = format("SELECT a, b, c FROM (%s) UNION ALL (%s)", subquery1, subquery2);
+
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, viewSql1, viewSql2);
+    }
+
+    @Test
+    public void multipleViewsSameBaseTableSomeCompatible()
+    {
+        String subquery1 = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String subquery2 = format("SELECT b, c FROM %s", BASE_TABLE_1);
+        String viewSql1 = subquery1;
+        String viewSql2 = format("SELECT c FROM %s", BASE_TABLE_1);
+
+        String baseQuerySql = format("SELECT a, b, c FROM (%s) UNION ALL (%s)", subquery1, subquery2);
+
+        String expectedRewrittenSql = format("SELECT a, b, c FROM (SELECT a, b FROM (%s)) UNION ALL (%s)",
+                VIEW_1, subquery2);
+
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, viewSql1, viewSql2);
+    }
```

Comment:
line break

Reply:
When you say line break for these, do you mean to change `assertOptimizedQuery` to be multiple lines, add an extra line of whitespace somewhere, or remove the existing line of whitespace?

Reply:
My bad, should of been clearer. Im saying to add the extra line break between the method and the next one for all occurrences as some have one between them and some don't

Reply:
Good catch, thanks!